### PR TITLE
Viewer data provider implementation in JavaScript

### PR DESCRIPTION
**Pull request by alex-deaconu**
- Use ES6 modules
- Incorporate GWT build processes in gulp
- Unit tests for Logger, utils, app and data-related modules
- Per-environment configuration injection
- Other minor fixes on build process

### CODE REVIEW

**Comment by alex-deaconu** [comment]
This is the first iteration of the Viewer re-write. A lot of stuff had to be moved (some even duplicated) into JS as it is needed for infrastructure. 

Also, some build process improvements have been incorporated into this branch as we were developing it.

@tejohnso please review. @ezequielc @settinghead FYI. Thanks!

File: src/main/java/com/risevision/viewer/client/channel/ChannelTokenProvider.java [hunk 1]
```diff
@@ -44,15 +39,15 @@ public void run() {
 	};
 	
 	public static void retrieveToken() {
-		if (ChannelConnectionController.isInactive() && !ViewerDataController.isBlocked()) {
```
Comment (ezequielc) [inline src/main/java/com/risevision/viewer/client/channel/ChannelTokenProvider.java]: @alex-deaconu Are we safe to remove this condition? Thanks

Comment (alex-deaconu) [inline src/main/java/com/risevision/viewer/client/channel/ChannelTokenProvider.java]: @ezequielc I believe so. Display blocking has been disabled in the Core for quite a while now. This is practically removing dead code. Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu It was hard for me to review it as I am not very familiar with Viewer code. I tested many scenarios and compared the code with GWT version. It all worked fine. We need to increase unit test coverage, but it is fine for the initial version. 👍  Cheers!

**Comment by alex-deaconu** [comment]
@ezequielc curious, did you review the Channel PR #116 as well?

Re: code, yes I agree not much to do other than compare to GWT code & test. As for test coverage, I agree it can be improved. 

Not sure if you saw my note re: coverage reporting in Hipchat Apps room. There's currently some issue as the reports are not very accurate. @settinghead will be looking into that. 

Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu No, I didn't review PR #116 yet. Will do now.
Yes, I saw your note re: coverage. Hopefully we get a much higher number. Cheers!

**Comment by settinghead** [comment]
@alex-deaconu @ezequielc I believe the coverage issue is now fixed (the issue was due to a misuse of a karma plugin).

Now statement coverage is at 98%, and function coverage around 90%, which I think is correct. I did a few tests to verify the metrics and the number appeared to increase/decrease according to my addition and deletion.

Also, uncovered lines seems to be reporting correctly (there are only 2 untested lines, which are in the Utils.js, and are two conditional branching which the test suite does not currently cover.

Please let me know if you still see any problems. Thanks!

**Comment by alex-deaconu** [comment]
@settinghead I'm not sure what the fix actually is. Can you please make a PR to the branch next time? It's easier to review, and I'm having to re-base all my other related branches if you force push. Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc @settinghead I've merged the other 2 PRs, rebased, and I believe all issues are resolved for this. Just want to confirm we are good to go. Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu 👍

**Comment by settinghead** [comment]
@alex-deaconu :+1:

**Comment by settinghead** [comment]
@alex-deaconu you may encounter a build error after merging this to master (eslint peer dependency issue). The solution is to clear cache and rebuild.

File: karma.config.js [hunk 2]
```diff
@@ -1,55 +1,75 @@
...
+          'karma-chai',
+          'sinon-chai',
+          'karma-chai-plugins',
+          'karma-browserify',
+          'karma-phantomjs-launcher',
+          'karma-coverage',
+          'karma-coveralls',
+      ],
+
+      singleRun: true,
```
Comment (settinghead) [inline karma.config.js]: @alex-deaconu can you remove this line before merging? Otherwise we'd have to make a new PR to fix it. Thanks.

**Pull request merged**